idb/postgres/postgres.go

```diff
@@ -2495,3 +2496,64 @@ func (db *IndexerDb) SetNetworkState(genesis bookkeeping.Genesis) error {
 	}
 	return db.setNetworkState(nil, &networkState)
 }
+
+// DeleteTransactions removes old transactions
+// keep is the number of rounds to keep in db
+func (db *IndexerDb) DeleteTransactions(ctx context.Context, keep uint64) (int64, error) {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Why do we create another context from the passed in context?

Reply:
updated.